src/System.Private.CoreLib/shared/System/Globalization/ISOWeek.cs

```diff
@@ -0,0 +1,135 @@
+        }
+
+        // From https://en.wikipedia.org/wiki/ISO_week_date#Weeks_per_year.
+        public static int GetWeeksInYear(int year)
+        {
+            if (year < MinYear || year > MaxYear)
+            {
+                throw new ArgumentOutOfRangeException(nameof(year), SR.ArgumentOutOfRange_Year);
+            }
+
+            int P(int y) => (y + (y / 4) - (y / 100) + (y / 400)) % 7;
+
+            if (P(year) == 4 || P(year - 1) == 3)
+            {
+                return WeeksInLongYear;
+            }
+
+            return WeeksInShortYear;
+        }
+
+        // From https://en.wikipedia.org/wiki/ISO_week_date#Calculating_a_date_given_the_year,_week_number_and_weekday:
+        //
+        // This method requires that one know the weekday of 4 January of the year in question.
+        // Add 3 to the number of this weekday, giving a correction to be used for dates within this year.
+        //
+        // Multiply the week number by 7, then add the weekday. From this sum subtract the correction for the year.
+        // The result is the ordinal date, which can be converted into a calendar date.
+        //
+        // If the ordinal date thus obtained is zero or negative, the date belongs to the previous calendar year.
+        // If greater than the number of days in the year, to the following year.
+        public static DateTime ToDateTime(int year, int week, DayOfWeek dayOfWeek)
+        {
+            if (year < MinYear || year > MaxYear)
+            {
+                throw new ArgumentOutOfRangeException(nameof(year), SR.ArgumentOutOfRange_Year);
+            }
+
+            if (week < MinWeek || week > MaxWeek)
+            {
+                throw new ArgumentOutOfRangeException(nameof(week), SR.ArgumentOutOfRange_Week_ISO);
+            }
+
+            if ((int)dayOfWeek < 0 || (int)dayOfWeek > 6)
+            {
+                throw new ArgumentOutOfRangeException(nameof(dayOfWeek), SR.ArgumentOutOfRange_DayOfWeek);
+            }
+
+            var jan4 = new DateTime(year, month: 1, day: 4);
+
+            int correction = GetWeekday(jan4.DayOfWeek) + 3;
+
+            int ordinal = (week * 7) + GetWeekday(dayOfWeek) - correction;
+                
+            return new DateTime(year, month: 1, day: 1).AddDays(ordinal - 1);
+        }
+
+        // From https://en.wikipedia.org/wiki/ISO_week_date#Calculating_the_week_number_of_a_given_date:
+        //
+        // Using ISO weekday numbers (running from 1 for Monday to 7 for Sunday),
+        // subtract the weekday from the ordinal date, then add 10. Divide the result by 7.
+        // Ignore the remainder; the quotient equals the week number.
+        //
+        // If the week number thus obtained equals 0, it means that the given date belongs to the preceding (week-based) year.
+        // If a week number of 53 is obtained, one must check that the date is not actually in week 1 of the following year.
```

Comment:
could you please copy these 2 lines comment to GetWeekOfYear method too? would be easier to read the code of GetWeekOfYear when having this comment there.

Reply:
This comment would be useful on GetYear method too

---
In reply to: [195190456](https://github.com/dotnet/coreclr/pull/18456#discussion_r195190456) [](ancestors = 195190456)